designer/server/src/main/scala/pl/touk/nussknacker/ui/api/DictResourcesEndpoints.scala

```diff
@@ -0,0 +1,140 @@
+package pl.touk.nussknacker.ui.api
+
+import io.circe.Json
+import io.circe.generic.JsonCodec
+import pl.touk.nussknacker.engine.api.dict.DictEntry
+import pl.touk.nussknacker.engine.api.process.ProcessingType
+import pl.touk.nussknacker.restmodel.BaseEndpointDefinitions
+import pl.touk.nussknacker.restmodel.BaseEndpointDefinitions.SecuredEndpoint
+import pl.touk.nussknacker.security.AuthCredentials
+import pl.touk.nussknacker.ui.api.DictResourcesEndpoints.DictError
+import pl.touk.nussknacker.ui.api.DictResourcesEndpoints.DictError.{MalformedTypingResult, NoDict, NoProcessingType}
+import pl.touk.nussknacker.ui.api.DictResourcesEndpoints.Dtos._
+import pl.touk.nussknacker.ui.api.description.TypingDtoSchemas
+import sttp.model.StatusCode.{BadRequest, NotFound, Ok}
+import sttp.tapir._
+import sttp.tapir.json.circe.jsonBody
+
+import scala.language.implicitConversions
+
+class DictResourcesEndpoints(auth: EndpointInput[AuthCredentials]) extends BaseEndpointDefinitions {
+
+  lazy val dictionaryEntryQueryEndpoint: SecuredEndpoint[(String, String, String), DictError, List[DictEntry], Any] =
+    baseNuApiEndpoint
+      .summary("Get list of dictionary entries matching the label pattern")
+      .tag("Dictionary")
+      .get
+      .in(
+        "processDefinitionData" / path[String]("processingType") / "dicts" / path[String]("dictId") / "entry" / query[
+          String
+        ]("label")
+      )
+      .out(
+        statusCode(Ok).and(
+          jsonBody[List[DictEntry]]
+        )
+      )
+      .errorOut(
+        oneOf[DictError](
+          oneOfVariantFromMatchType(
+            NotFound,
+            plainBody[NoProcessingType]
+          ),
+          oneOfVariantFromMatchType(
+            NotFound,
+            plainBody[NoDict]
+          )
+        )
+      )
+      .withSecurity(auth)
+
+  lazy val dictionaryListEndpoint
+      : SecuredEndpoint[(String, DictListRequestDto), DictError, List[DictListElementDto], Any] =
+    baseNuApiEndpoint
+      .summary("Get list of available dictionaries with value type compatible with expected type")
+      .tag("Dictionary")
+      .post
+      .in("processDefinitionData" / path[String]("processingType") / "dicts")
+      .in(jsonBody[DictListRequestDto])
+      .out(
+        statusCode(Ok).and(
+          jsonBody[List[DictListElementDto]]
+        )
+      )
+      .errorOut(
+        oneOf[DictError](
+          oneOfVariantFromMatchType(
+            NotFound,
+            plainBody[NoProcessingType]
+          ),
+          oneOfVariantFromMatchType(
+            BadRequest,
+            plainBody[MalformedTypingResult]
+          )
+        )
+      )
+      .withSecurity(auth)
+
+}
+
+object DictResourcesEndpoints {
+
+  object Dtos {
+
+    @JsonCodec
+    case class TypingResultInJson(value: Json)
+
+    @JsonCodec
+    case class DictListRequestDto(expectedType: TypingResultInJson)
+
+    @JsonCodec
+    case class DictListElementDto(
```

Comment:
Element of list of dicts is just a dict! :) We can call it `DictDto`. `dictId` is just `id`. It is a quite simple CRUD at the end.